### PR TITLE
JournalArticle ContentType with tests

### PR DESCRIPTION
**Pull request by svenngjeto**

### CODE REVIEW

File: src/main/java/no/unit/nva/model/instancetypes/journal/JournalArticleContentType.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    REVIEW_ARTICLE("Review article"),
+    CASE_REPORT("Case report"),
+    STUDY_PROTOCOL("Study protocol"),
+    PROFESSIONAL_ARTICLE("Professional article"),
+    POPULAR_SCIENCE_ARTICLE("Popular science article");
+
+    public static final String ERROR_MESSAGE_TEMPLATE = "%s not a valid JournalArticleContentType, expected one of: %s";
+    public static final String DELIMITER = ", ";
+
+    private String value;
```
Comment (axthosarouris) [inline src/main/java/no/unit/nva/model/instancetypes/journal/JournalArticleContentType.java:20]: final

File: src/test/java/no/unit/nva/model/instancetypes/journal/JournalArticleTest.java [hunk 2]
```diff
@@ -18,14 +27,79 @@ void journalArticleReturnsObjectWhenPeerReviewIsTrueOrFalse(boolean peerReviewed
...
+        JournalArticle actualJournalArticle = objectMapper.readValue(expectedJson, JournalArticle.class);
+        assertEquals(expectedJournalArticle, actualJournalArticle);
+
+        String actualJson = objectMapper.writeValueAsString(expectedJournalArticle);
+        assertEquals(expectedJson, actualJson);
+    }
+
+    @Test
+    public void canSerializeJournalArticleWithContentTypeWithoutDataloss() throws JsonProcessingException {
+        String contentTypeString = "Research article";
```
Comment (axthosarouris) [inline src/test/java/no/unit/nva/model/instancetypes/journal/JournalArticleTest.java:73]: Use the enum and the getValue() method: `JournalArticleContentType.RESEARCH_ARTICLE.getValue();`

File: src/test/java/no/unit/nva/model/instancetypes/journal/JournalArticleTest.java [hunk 3]
```diff
@@ -18,14 +27,79 @@ void journalArticleReturnsObjectWhenPeerReviewIsTrueOrFalse(boolean peerReviewed
...
+        JournalArticle expectedJournalArticle = generateJournalArticle(expectedJournalArticleTestData);
+        String expectedJson = objectMapper.writeValueAsString(expectedJournalArticle);
+        JournalArticle actualJournalArticle = objectMapper.readValue(expectedJson, JournalArticle.class);
+
+        assertEquals(expectedJournalArticle, actualJournalArticle);
+        assertThat(expectedJournalArticle, is(equalTo(actualJournalArticle)));
+    }
+
+    @Test
+    public void journalArticleBuilderCreatesObjectWithoutEmptyValues() throws JsonProcessingException {
```
Comment (axthosarouris) [inline src/test/java/no/unit/nva/model/instancetypes/journal/JournalArticleTest.java:86]: ...ReturnsObject

File: src/test/java/no/unit/nva/model/instancetypes/journal/JournalArticleTest.java [hunk 4]
```diff
@@ -18,14 +27,79 @@ void journalArticleReturnsObjectWhenPeerReviewIsTrueOrFalse(boolean peerReviewed
...
+        String contentTypeString = "Research article";
```
Comment (axthosarouris) [inline src/test/java/no/unit/nva/model/instancetypes/journal/JournalArticleTest.java:87]: use the enum and the getValue method.

File: src/test/java/no/unit/nva/model/instancetypes/journal/JournalArticleTest.java [hunk 5]
```diff
@@ -18,14 +27,79 @@ void journalArticleReturnsObjectWhenPeerReviewIsTrueOrFalse(boolean peerReviewed
...
+
+        String actualJson = objectMapper.writeValueAsString(expectedJournalArticle);
+        assertEquals(expectedJson, actualJson);
+    }
+
+    @Test
+    public void canSerializeJournalArticleWithContentTypeWithoutDataloss() throws JsonProcessingException {
+        String contentTypeString = "Research article";
+        JournalTestData expectedJournalArticleTestData = new JournalTestData();
+        expectedJournalArticleTestData.setContent(contentTypeString);
```
Comment (axthosarouris) [inline src/test/java/no/unit/nva/model/instancetypes/journal/JournalArticleTest.java:75]: Should we add this value to the contructor of the JournalTestData?

File: src/test/java/no/unit/nva/model/instancetypes/journal/JournalArticleTest.java [hunk 6]
```diff
@@ -18,14 +27,79 @@ void journalArticleReturnsObjectWhenPeerReviewIsTrueOrFalse(boolean peerReviewed
         assertDoesNotThrow(() -> generateJournalArticle(testData));
     }
 
+    @Test
```
Comment (axthosarouris) [inline src/test/java/no/unit/nva/model/instancetypes/journal/JournalArticleTest.java:30]: We also need a test that we test that the serialization form is `"content" : <string>`

**Review by axthosarouris — CHANGES_REQUESTED** [review]

File: src/main/java/no/unit/nva/model/instancetypes/journal/JournalArticleContentType.java (hunk 1 above)
Comment (svenngjeto) [reply src/main/java/no/unit/nva/model/instancetypes/journal/JournalArticleContentType.java:20]: Final!

File: src/test/java/no/unit/nva/model/instancetypes/journal/JournalArticleTest.java (hunk 5 above)
Comment (svenngjeto) [reply src/test/java/no/unit/nva/model/instancetypes/journal/JournalArticleTest.java:75]: Yes, looks better

File: src/test/java/no/unit/nva/model/instancetypes/journal/JournalArticleTest.java (hunk 6 above)
Comment (svenngjeto) [reply src/test/java/no/unit/nva/model/instancetypes/journal/JournalArticleTest.java:33]: Made a simple one..

**Review by axthosarouris — APPROVED** [review]

**Comment by axthosarouris** [comment]
Just change the CharSequence witth String?

**Pull request merged**